src/mixins/itext_key_behavior.mixin.js

```diff
@@ -4,12 +4,17 @@ fabric.util.object.extend(fabric.IText.prototype, /** @lends fabric.IText.protot
    * Initializes hidden textarea (needed to bring up keyboard in iOS)
    */
   initHiddenTextarea: function() {
+    var text = this.canvas.getActiveObject();
+    // Position in the middle because we don't know where the cursor is
+    var left = text.oCoords.ml.x;
```

Comment:
instead of text.oCoords.ml.x use:
var rect = this.getBoundingRect(), left = rect.left, top = rect.top + rect.height;
In this way you have always medium left corner, also when the text is rotated or flipped.